lib/src/api/err/mod.rs

```diff
@@ -181,6 +181,18 @@ pub enum Error {
 	/// Tried to use a range query on an edge or edges
 	#[error("Live queries on edges not supported: {0}")]
 	LiveOnEdges(Edges),
```

Comment:
I was going to comment on the one below, but the one below is actually a query issue not an LQ issue.
Is the `LiveOnEdges` error already committed to the API? Because a better place for it would be a Live Query error which I believe already exists.

Reply:
> I was going to comment on the one below, but the one below is actually a query issue not an LQ issue.

The one below is returned when someone uses `Response::stream` instead of `Response::take` on a normal query.

> Is the LiveOnEdges error already committed to the API?

Yes, this was introduced in the live query API PR so it's already part of the public API since v1.1.0.

> a better place for it would be a Live Query error which I believe already exists.

I'm not sure which one you are referring to specifically. This is purely an API error though. It happens when someone uses an `Edges` object as a parameter to `Surreal::select` and subsequently call `live` on that.

```rust
db.select(Edges { ... }).live().await?;
```

It's [returned here](https://github.com/surrealdb/surrealdb/blob/f587289923ca261c07f4677e3c63a51454d8cc60/lib/src/api/method/live.rs#L87-L89). In future we may be able to turn this into a compile time error and deprecate those variants.

Reply:
Yeah fine lets keep it as is. It would be convenient to have a separation of syntax errors, vs logic errors, vs internal errors. As it can give an indication if something is retriable and under what circumstances.

Reply:
I think we already have that, to some extend.

```rust
pub enum Error {
    Db(Db),
    Api(Api),
}
```

I think both contain internal errors though.